<?xml version="1.0"?>
<launch>
    <!-- TODO: Populate doc -->
    <arg name="config_file" default="files/configuration.yaml" doc=""/>
    <arg name="generate_samples" default="True" doc="Generate or not the initial sampling of space."/>
    <arg name="filter_reachable" default="True" doc="Filter the initial sampling : test is each pose can be achieved by the robot"/>
    <arg name="order_reachable" default="True" doc="Order the reachable poses : order them so that the n first are as far a possible from each other"/>
    <arg name="export_poses" default="True" doc="Export the n first reachable pose in separate files"/>

    <!-- Argument to enable plotting -->
    <arg name="plot" default="True" doc="Enable plotting the poses"/>
    <arg name="plot_samples" default="True" doc=""/>
    <arg name="plot_reachables" default="True" doc=""/>
    <arg name="plot_finals" default="True" doc=""/>

    <include file="$(find prl_pinocchio)/launch/upload_ur5.launch"/>

    <!-- Run calibration node -->
    <node pkg="prl_ur5_calibration" type="generate_poses.py" name="generate_poses" output="screen">
        <param name="config_file"       value="$(arg config_file)"/>
        <param name="generate_samples"  value="$(arg generate_samples)"/>
        <param name="filter_reachable"  value="$(arg filter_reachable)"/>
        <param name="order_reachable"   value="$(arg order_reachable)"/>
        <param name="export_poses"      value="$(arg export_poses)"/>
        <param name="plot"              value="$(arg plot)"/>
        <param name="plot_samples"      value="$(arg plot_samples)"/>
        <param name="plot_reachables"   value="$(arg plot_reachables)"/>
        <param name="plot_finals"       value="$(arg plot_finals)"/>
    </node>
</launch>
